### DISCUSSION TITLE
Configurable Velocity disconnect message

### DISCUSSION THREAD

**Discussion by Vrganj** [PaperMC/Paper#10509] [Ideas]
### Is your feature request related to a problem?

Players sometimes connect to an old IP they found in chat and get met with "This server requires you to connect with Velocity."
I'd like to configure the message, as very little players know what that means.

### Describe the solution you'd like.

I suggest a new field in the `messages` section of paper-global.yml that makes this configurable.
The code containing the message is found in `0774-Add-Velocity-IP-Forwarding-Support.patch`

### Describe alternatives you've considered.

I thought of using AsyncPlayerPreLoginEvent, but it seems that the disconnect happens before it's even called.

**Comment by electronicboy** [comment]
It's somewhat intentional that we don't allow replacing messages which are generally only visible due to lower than recommended security setups

**Comment by Vrganj** [comment]
I'm trying to avoid losing old players who had the IP saved before we even setup a proxy. Why would you consider this setup "lower than recommended"?

**Comment by electronicboy** [comment]
Because your backend instances should generally not be exposed to the public internet, velocity makes it safer but it is still a security risk, especially if you're running 3rd party plugins, etc; The recommended solution for picking up a proxy is in "shared hosting envs" is to put the proxy where your old ingress was, and move the backend server to somewhere else so you can retain the old ip/port combo

**Comment by mitchellcrazyeye** [comment]
I'll +1 this - I somewhat disagree with the concept of how this was interpreted. I don't use the proxy for security - I use it so players don't get disconnected. This is on two different servers, one remote, one local - the IP has to be public otherwise the proxy can't connect. Sure, I could setup firewall rules that _only_ let the proxy connect, but this is also on a home internet connection - it's not based in a datacenter with all of those tools in front of me. Players are running into this error when trying to connect via the old IP - this should be customizable.

The TL;DR of this whole thing is that proxies have different use cases outside of just security and sometimes things happen outside of our control / need to happen to tackle an issue headon and you can't keep your old IP/Port combo on the proxy - or in my case, I moved the proxy to a remote, more stable environment for cheap to keep players online when needed.

**Reply by electronicboy** [reply]
exposing the backend servers incurs a risk, especially when using less secure forwarding modes. For velocity forwarding this is generally not as huge of a risk given the secret, but, it's generally not a use case of people leaving this stuff exposed to the web that we want to support.

I'm not 100% against having a config message for velocity mode (and only velocity mode), but, leaving your backends exposed to the public was not exactly a "we should cater for people doing this" outside of providing a useful warning other than just terminating the connection with nothing identifiable to support.

**Reply by mitchellcrazyeye** [reply]
Yeah, I understand it's not the perfect setup - but, if an end-user sees it, it would be nice (with acknowledgement of risk, maybe above the config option) to be able to change this message just to say "connect via mc.website.com" rather than Velocity, sparking a question to us as admins.

Thanks for hearing me out
